Bluejay.asm

```diff
@@ -3482,29 +3507,38 @@ erase_and_store_all_in_eeprom:
 	mov	A, #EEPROM_LAYOUT_REVISION
 	call	write_eeprom_byte_from_acc
 
-	; Write eeprom
+	; Write eeprom before Eep_Initialized_x flags
+	; Temp4 = 10 parameters [_Eep_Pgm_Gov_P_Gain - Eep_Initialized_x]
 	mov	DPTR, #_Eep_Pgm_Gov_P_Gain
 	mov	Temp1, #_Pgm_Gov_P_Gain
-	mov	Temp4, #10				; 10 parameters
+	mov	Temp4, #10
 write_eeprom_block1:
 	call	write_eeprom_byte
 	inc	DPTR
 	inc	Temp1
 	djnz	Temp4, write_eeprom_block1
 
+	; Write eeprom after Eep_Initialized_x flags
+	; Temp4 = 27 parameters: [_Eep_Enable_TX_Program - Eep_Pgm_Power_Rating]
 	mov	DPTR, #_Eep_Enable_TX_Program
 	mov	Temp1, #_Pgm_Enable_TX_Program
-	mov	Temp4, #26				; 26 parameters
+	mov	Temp4, #27
 write_eeprom_block2:
 	call	write_eeprom_byte
 	inc	DPTR
 	inc	Temp1
 	djnz	Temp4, write_eeprom_block2
 
+	; Now write tags, melogy and signature
 	call	write_tags
 	call	write_melody
 	call	write_eeprom_signature
 	mov	DPTR, #Eep_Dummy			; Set pointer to uncritical area
+
+	; Beep signal to confirm action done and giving time to flash controller to settle data down
```

Comment:
I am not entirely sure how I feel about that. The beep codes are already so ingrained in peoples minds, I am afraid that we will have a lot of people wondering why their ESC is now making additional sounds.

Have you found that the 200ms wait is actually needed for the data to be correctly written? If so, I would keep the timeout but don't have it beep.

Maybe I am too paranoid about how users will react - just food for thought.

Reply:
I know that it microcontroller resets (for any reason) before flash controller can settle data down, data would corrupt. If you like we could remove the beeps, but I think we could keep the 200ms wait. In the worst case it will do nothing, and nobody would realize.

Reply:
Good point, yes - that makes sense then to keep the wait - perfect.